File: MesWebSite/WebUI/WebSiteScripts/QcmQaRecord.js
$(function () {
    //绑定上侧数据
    bindTopData();
    //初始化上侧表格
    initialTblTop();
    //上侧刷新按钮单击事件
    $('#TopToolBtnRefresh').click(function () {
        topTblRefresh();
    });
    ////上侧编辑按钮单击事件
    //$('#TopToolBtnEdit').click(function () {
    //    topEditClick();
    //});
    ////上侧添加按钮单击事件
    //$('#TopToolBtnAdd').click(function () {
    //    topAddClick();
    //});
    //上侧删除按钮单击事件
    $('#TopToolBtnDelete').click(function () {
        topDeleteClick();
    });
});
//绑定上侧数据
function bindTopData() {
    bindStatusTop5();
    bindEqm();
    bindQaCause();
    bindQaResult();
}
//初始化上侧表格
function initialTblTop() {
    $('#TblTop').datagrid({
        url: '/BackgroundProgram/GetListPage.ashx',
        title: '质量异常记录',
        fitColumns: false,
        idField: 'qa_record_no',
        loadMsg: '正在获取后台信息...',
        fit: true,
        pagination: true,
        singleSelect: true,
        pageSize: 10,
        pageNumber: 1,
        pageList: [10, 20, 30],
        queryParams: { viewNeedOperate: 'qcmQaRecord', orderBy: JSON.stringify({ qa_record_no: 'desc' }), where: strBase64('qa_result_no is not null') },
        columns: [[
            { field: 'qa_record_no', title: '异常编号', align: 'left', width: 140 },
                    { field: 'status_no', title: '状态码', align: 'left', width: 140 },
                    { field: 'status_name', title: '状态', align: 'left', width: 140 },
                    { field: 'qa_cause_no', title: '异常代码', align: 'left', width: 140 },
                    { field: 'qa_cause_name', title: '异常原因', align: 'left', width: 140 },
                    { field: 'part_no', title: '品号', align: 'left', width: 140 },
                    { field: 'eqm_no', title: '设备编号', align: 'left', width: 140 },
                    { field: 'mpo_no', title: '生产单号', align: 'left', width: 140 },
                    { field: 'mpo_item_no', title: '子单编号', align: 'left', width: 140 },
                    { field: 'serial_no', title: '序列号', align: 'left', width: 140 },
                    { field: 'qa_result_no', title: '结果编号', align: 'left', width: 140 },
                    { field: 'qa_result_name', title: '质量结果', align: 'left', width: 140 },
                    //{ field: 'submit_user_no', title: '上报人编号', align: 'left', width: 140 },
                    //{ field: 'submit_user_name', title: '上报人', align: 'left', width: 140 },
                    { field: 'submit_time', title: '上报时间', align: 'left', width: 140 },
                    //{ field: 'solve_user_no', title: '处理人编号', align: 'left', width: 140 },
                    //{ field: 'solve_user_name', title: '处理人', align: 'left', width: 140 },
                    { field: 'solve_time', title: '处理时间', align: 'left', width: 140 }
        ]],
        toolbar: '#ToolbarTop'
    });
}

//上侧数据刷新
function topTblRefresh() {
    $('#TblTop').datagrid('clearSelections').datagrid('clearChecked');
    $('#TblTop').datagrid('reload');
}

//上侧编辑按钮单击事件
function topEditClick() {
    var row = $('#TblTop').datagrid('getSelected');
    if (row === null) {
        $.messager.alert('警告', '未选中要修改的对象', 'warning');
        return;
    }
    $('#TxtQaRecordNo').textbox('setText', row.qa_record_no);
    $('#TxtQaRecordNo').textbox('disable');
    $('#CmbStatusTop').combobox('select', row.status_no);
    $('#TxtPartNo').textbox('setText', row.part_no);
    $('#TxtMpoNo').textbox('setText', row.mpo_no);
    $('#TxtMpoItemNo').textbox('setText', row.mpo_item_no);
    $('#TxtSerialNo').textbox('setText', row.serial_no);
    $('#CmbEqm').combobox('select', row.eqm_no);
    $('#CmbQaCause').combobox('select', row.qa_cause_no);
    $('#CmbQaResult').combobox('select', row.qa_result_no);
    $('#TopEditFrm').dialog({
        title: '添加/编辑窗口',
        modal: true,
        collapsible: true,
        minizable: true,
        maxizable: true,
        resizeable: true,
        buttons: [{
            text: '确认',
            handler: function () {
                row.qa_record_no = $('#TxtQaRecordNo').textbox('getText');
                row.status_no = $('#CmbStatusTop').combobox('getValue');
                row.status_name = $('#CmbStatusTop').combobox('getText');
                row.part_no = $('#TxtPartNo').textbox('getText');
                row.mpo_no = $('#TxtMpoNo').textbox('getText');
                row.mpo_item_no = $('#TxtMpoItemNo').textbox('getText');
                row.serial_no=$('#TxtSerialNo').textbox('getText');
                row.eqm_no = $('#CmbEqm').combobox('getValue');
                row.eqm_name= $('#CmbEqm').combobox('getText');
                row.qa_cause_no = $('#CmbQaCause').combobox('getValue');
                row.qa_cause_name = $('#CmbQaCause').combobox('getText');
                row.qa_result_no = $('#CmbQaResult').combobox('getValue');
                row.qa_result_name = $('#CmbQaResult').combobox('getText');
                $.post('/BackgroundProgram/EditObject.ashx', { viewNeedOperate: 'qcmQaRecord', objNeedOperate: JSON.stringify(row) }, function (data) {
                    if (data === null) {
                        $.messager.alert('提示信息', '无法获取结果！', 'error');
                        return;
                    }
                    var msg = JSON.parse(data).msg;
                    if (msg === 'success') {
                        topTblRefresh();
                        $('#TopEditFrm').dialog('close');
                        $('#TxtQaRecordNo').textbox('enable');
                    }
                    else {
                        $.messager.alert('提示信息', msg, 'error');
                    }
                });
            }
        }, {
            text: '取消',
            handler: function () {
                $('#TopEditFrm').dialog('close');
                $('#TxtQaRecordNo').textbox('enable');
            }
        }],
        onClose: function () {
            $('#TxtQaRecordNo').textbox('enable');
        }
    });
}

//上侧添加按钮单击事件
function topAddClick() {
    $('#TopEditFrm').dialog({
        title: '添加/编辑窗口',
        modal: true,
        collapsible: true,
        minizable: true,
        maxizable: true,
        resizeable: true,
        buttons: [{
            text: '确认',
            handler: function () {
                var qa_record_no = $('#TxtQaRecordNo').textbox('getText');
                var status_no = $('#CmbStatusTop').combobox('getValue');
                var status_name = $('#CmbStatusTop').combobox('getText');
                var part_no = $('#TxtPartNo').textbox('getText');
                var mpo_no = $('#TxtMpoNo').textbox('getText');
                var mpo_item_no = $('#TxtMpoItemNo').textbox('getText');
                var serial_no = $('#TxtSerialNo').textbox('getText');
                var eqm_no = $('#CmbEqm').combobox('getValue');
                var eqm_name = $('#CmbEqm').combobox('getText');
                var qa_cause_no = $('#CmbQaCause').combobox('getValue');
                var qa_cause_name = $('#CmbQaCause').combobox('getText');
                var qa_result_no = $('#CmbQaResult').combobox('getValue');
                var qa_result_name = $('#CmbQaResult').combobox('getText');
                $.post('/BackgroundProgram/AddObject.ashx', { viewNeedOperate: 'qcmQaRecord', objNeedOperate: JSON.stringify({ qa_record_no: qa_record_no, status_no: status_no, status_name: status_name, part_no: part_no, mpo_no: mpo_no, mpo_item_no: mpo_item_no, serial_no: serial_no, eqm_no: eqm_no, eqm_name: eqm_name, qa_cause_no: qa_cause_no, qa_cause_name: qa_cause_name, qa_result_no: qa_result_no, qa_result_name: qa_result_name }) }, function (data) {
                    if (data === null) {
                        $.messager.alert('提示信息', '无法获取结果！', 'error');
                        return;
                    }
                    var msg = JSON.parse(data).msg;
                    if (msg === 'success') {
                        topTblRefresh();
                        $('#TopEditFrm').dialog('close');
                    }
                    else {
                        $.messager.alert('提示信息', msg, 'error');
                    }
                });
            }
        }, {
            text: '取消',
            handler: function () {
                $('#TopEditFrm').dialog('close');
            }
        }]
    });
}
//上侧删除按钮单击事件
function topDeleteClick() {
    var rows = $('#TblTop').datagrid('getSelections');
    if (rows.length <= 0) {
        $.messager.alert('警告', '未选中要删除的对象', 'warning');
        return;
    }
    $.messager.confirm('确认', '确认要删除这些数据？', function (r) {
        if (!r) {
            return;
        }
        $.post('/BackgroundProgram/DeleteObject.ashx', { viewNeedOperate: 'qcmQaRecord', objNeedOperate: JSON.stringify(rows) }, function (data) {
            if (data === null) {
                $.messager.alert('提示信息', '无法获取结果！', 'error');
                return;
            }
            var msg = JSON.parse(data).msg;
            if (msg === 'success') {
                topTblRefresh();
            }
            else {
                $.messager.alert('提示信息', msg, 'error');
            }
        });
    });
}

File: MesLib/Model/TableModel/Qcm_qa_record.cs
/**  版本信息模板在安装目录下，可自行修改。
* Qcm_qa_record.cs
*
* 功 能： N/A
* 类 名： Qcm_qa_record
*
* Ver    变更日期             负责人  变更内容
* ───────────────────────────────────
* V0.01  2018/7/24 10:26:49   N/A    初版
*
* Copyright (c) 2012 Maticsoft Corporation. All rights reserved.
*┌──────────────────────────────────┐
*│　此技术信息为本公司机密信息，未经本公司书面同意禁止向第三方披露．　│
*│　版权所有：动软卓越（北京）科技有限公司　　　　　　　　　　　　　　│
*└──────────────────────────────────┘
*/
using System;
namespace Model.TableModel
{
	/// <summary>
	/// Qcm_qa_record:实体类(属性说明自动提取数据库字段的描述信息)
	/// </summary>
	[Serializable]
	public partial class Qcm_qa_record
	{
		public Qcm_qa_record()
		{}
		#region Model
		private string _id;
		private string _status_no;
		private string _status_name;
		private string _crt_user_no;
		private string _crt_user_name;
		private DateTime _crt_time;
		private string _upd_user_no;
		private string _upd_user_name;
		private DateTime _upd_time;
		private string _qa_record_no;
		private string _qa_cause_no;
		private string _qa_cause_name;
		private string _part_no;
		private string _eqm_no;
		private string _mpo_no;
		private string _mpo_item_no;
		private string _serial_no;
		private string _factory_no;
		private string _submit_user_no;
		private string _submit_user_name;
		private DateTime _submit_time;
		private string _qa_result_no;
		private string _qa_result_name;
		private string _solve_user_no;
		private string _solve_user_name;
		private DateTime? _solve_time;
		/// <summary>
		/// 
		/// </summary>
		public string id
		{
			set{ _id=value;}
			get{return _id;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string status_no
		{
			set{ _status_no=value;}
			get{return _status_no;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string status_name
		{
			set{ _status_name=value;}
			get{return _status_name;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string crt_user_no
		{
			set{ _crt_user_no=value;}
			get{return _crt_user_no;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string crt_user_name
		{
			set{ _crt_user_name=value;}
			get{return _crt_user_name;}
		}
		/// <summary>
		/// 
		/// </summary>
		public DateTime crt_time
		{
			set{ _crt_time=value;}
			get{return _crt_time;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string upd_user_no
		{
			set{ _upd_user_no=value;}
			get{return _upd_user_no;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string upd_user_name
		{
			set{ _upd_user_name=value;}
			get{return _upd_user_name;}
		}
		/// <summary>
		/// 
		/// </summary>
		public DateTime upd_time
		{
			set{ _upd_time=value;}
			get{return _upd_time;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string qa_record_no
		{
			set{ _qa_record_no=value;}
			get{return _qa_record_no;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string qa_cause_no
		{
			set{ _qa_cause_no=value;}
			get{return _qa_cause_no;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string qa_cause_name
		{
			set{ _qa_cause_name=value;}
			get{return _qa_cause_name;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string part_no
		{
			set{ _part_no=value;}
			get{return _part_no;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string eqm_no
		{
			set{ _eqm_no=value;}
			get{return _eqm_no;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string mpo_no
		{
			set{ _mpo_no=value;}
			get{return _mpo_no;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string mpo_item_no
		{
			set{ _mpo_item_no=value;}
			get{return _mpo_item_no;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string serial_no
		{
			set{ _serial_no=value;}
			get{return _serial_no;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string factory_no
		{
			set{ _factory_no=value;}
			get{return _factory_no;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string submit_user_no
		{
			set{ _submit_user_no=value;}
			get{return _submit_user_no;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string submit_user_name
		{
			set{ _submit_user_name=value;}
			get{return _submit_user_name;}
		}
		/// <summary>
		/// 
		/// </summary>
		public DateTime submit_time
		{
			set{ _submit_time=value;}
			get{return _submit_time;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string qa_result_no
		{
			set{ _qa_result_no=value;}
			get{return _qa_result_no;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string qa_result_name
		{
			set{ _qa_result_name=value;}
			get{return _qa_result_name;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string solve_user_no
		{
			set{ _solve_user_no=value;}
			get{return _solve_user_no;}
		}
		/// <summary>
		/// 
		/// </summary>
		public string solve_user_name
		{
			set{ _solve_user_name=value;}
			get{return _solve_user_name;}
		}
		/// <summary>
		/// 
		/// </summary>
		public DateTime? solve_time
		{
			set{ _solve_time=value;}
			get{return _solve_time;}
		}
		#endregion Model

	}
}

